include/deal.II/fe/fe_nothing.h

```diff
@@ -82,10 +82,12 @@ class FE_Nothing : public FiniteElement<dim,spacedim>
 public:
 
   /**
-   * Constructor. Argument denotes the number of components to give this
-   * finite element (default = 1).
+   * Constructor. First argument denotes the number of components to give this
+   * finite element (default = 1). Second argument decides whether FE_Nothing
+   * will dominate any other FE in compare_for_face_domination() (default = false).
    */
-  FE_Nothing (const unsigned int n_components = 1);
+  FE_Nothing (const unsigned int n_components = 1,
+              const bool dominate = false);
```

Comment:
On 08/26/2015 06:59 AM, Denis Davydov wrote:
> In include/deal.II/fe/fe_nothing.h
> <https://github.com/dealii/dealii/pull/1424#discussion_r37973319>:
>
>>     */
>> -  FE_Nothing (const unsigned int n_components = 1);
>> +  FE_Nothing (const unsigned int n_components = 1,
>> +              const bool dominate = false);
>
>     I don't think so.
>
> Apparently that's not the case:
>
> |   deallog << "FESystem:"<<std::endl;
>    Triangulation<dim>   triangulation;
>    {
>      Triangulation<dim>   triangulationL;
>      Triangulation<dim>   triangulationR;
>      GridGenerator::hyper_cube (triangulationL, -1,0); //create a square [-1,0]^d domain
>      GridGenerator::hyper_cube (triangulationR, -1,0); //create a square [-1,0]^d domain
>      Point<dim> shift_vector;
>      shift_vector[0] = 1.0;
>      GridTools::shift(shift_vector,triangulationR);
>      GridGenerator::merge_triangulations (triangulationL, triangulationR, triangulation);
>    }
>
>    hp::DoFHandler<dim> dof_handler(triangulation);
>    EnrichmentFunction<dim> function;
>
>    hp::FECollection<dim> fe_collection;
>    fe_collection.push_back(FESystem<dim>(FE_Q<dim>(1),1,
>                                          FE_Nothing<dim>(),1));
>    fe_collection.push_back(FESystem<dim>(FE_Q<dim>(2),1,
>                                          FE_Q<dim>(1),1));
>
>    dof_handler.begin_active()->set_active_fe_index(1);
>
>    dof_handler.distribute_dofs(fe_collection);
>
>    ConstraintMatrix constraints;
>    constraints.clear();
>    dealii::DoFTools::make_hanging_node_constraints  (dof_handler, constraints);
>    constraints.close ();
>
>    constraints.print(deallog.get_file_stream());
>
>    dof_handler.clear();
> |
>
> gives
>
> |DEAL::FESystem:
>      2 = 0
>      5 = 0
>      7 11:  0.500000
>      7 13:  0.500000
> |
>
> don't know whether this is an intended purpose, but that's how FESystem
> behaves now.

No, that's a bug. I've put the testcase into
   https://github.com/dealii/dealii/pull/1495

Will open an issue soon.